myth

```diff
@@ -177,147 +170,126 @@ if args.solv:
     else:
         if util.solc_exists(version):
             logging.info('Given version is already installed')
-            solc_binary = os.path.join(os.environ['HOME'], ".py-solc/solc-v" + version, "bin/solc")
-            logging.info("Setting the compiler to " + str(solc_binary))
         else:
             try:
                 solc.install_solc('v' + version)
-                solc_binary = os.path.join(os.environ['HOME'], ".py-solc/solc-v" + version, "bin/solc")
-                logging.info("Setting the compiler to " + str(solc_binary))
             except SolcError:
                 exitWithError(args.outform, "There was an error when trying to install the specified solc version")
+
+        solc_binary = os.path.join(os.environ['HOME'], ".py-solc/solc-v" + version, "bin/solc")
+        logging.info("Setting the compiler to " + str(solc_binary))
 else:
-    if not args.solv:
-        try:
-            solc_binary = os.environ['SOLC']
-        except KeyError:
-            try:
-                solc_binary = 'solc'
-            except:
-                exitWithError('No solidity compiler found, please make sure it is installed or specify it manually.')
+    try:
+        solc_binary = os.environ['SOLC']
+    except KeyError:
+        solc_binary = 'solc'
 
 # Establish RPC/IPC connection if necessary
 
-if (args.address or args.init_db):
-
-        if args.infura_mainnet:
-            eth = EthJsonRpc('mainnet.infura.io', 443, True)
-        elif args.infura_rinkeby:
-            eth = EthJsonRpc('rinkeby.infura.io', 443, True)
-        elif args.infura_kovan:
-            eth = EthJsonRpc('kovan.infura.io', 443, True)
-        elif args.infura_ropsten:
-            eth = EthJsonRpc('ropsten.infura.io', 443, True)
-        elif args.ganache:
-            eth = EthJsonRpc('localhost', 7545, False)
-        elif args.rpc:
-
-                try:
-                    host, port = args.rpc.split(":")
-                except ValueError:
-                    exitWithError(args.outform, "Invalid RPC argument, use HOST:PORT")
-
-                tls = args.rpctls
-
-                eth = EthJsonRpc(host, int(port), tls)
+eth = None
+if args.address or args.init_db:
+    if args.infura_mainnet:
+        eth = EthJsonRpc('mainnet.infura.io', 443, True)
+    elif args.infura_rinkeby:
+        eth = EthJsonRpc('rinkeby.infura.io', 443, True)
+    elif args.infura_kovan:
+        eth = EthJsonRpc('kovan.infura.io', 443, True)
+    elif args.infura_ropsten:
+        eth = EthJsonRpc('ropsten.infura.io', 443, True)
+    elif args.ganache:
+        eth = EthJsonRpc('localhost', 7545, False)
+    elif args.rpc:
+        try:
+            host, port = args.rpc.split(":")
+        except ValueError:
+            exitWithError(args.outform, "Invalid RPC argument, use HOST:PORT")
         else:
```

Comment:
Add a `else` to `try/except`, make the use of `host` and `port` in correct scope